version: '3.8'
services:
  db:
    image: postgres
    container_name: postgres_auth
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_PASSWORD=123qwe
      - POSTGRES_USER=user
      - POSTGRES_DB=users_jwt_base
    env_file:
      - .env
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 5s
      timeout: 5s
      retries: 5
  redis:
    image: redis:latest
    ports:
      - "6379:6379"
  nginx:
    image: nginx:latest
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro
      - ./nginx/sites.conf:/etc/nginx/conf.d/default.conf:ro
    depends_on:
      - flask
    ports:
      - "8001:80"
  jaeger:
    image: jaegertracing/all-in-one:latest
    ports:
      - "6831:6831/udp"
      - "16686:16686"
  flask:
    build:
      context: .
    depends_on:
      db:
        condition: service_healthy